### PR TITLE
Split DS Image layout VU and add missing attachment layout VU

### PR DESCRIPTION
**Pull request by spencer-lunarg**
- Moves the code to the source files instead of headers
- Adds all the missing `VUID-VkAttachmentDescription2-*` that map to a `VUID-VkAttachmentDescription-*`
- label `VUID-VkAttachmentDescription2-format-06247` and `VUID-VkAttachmentDescription2-format-06248`
- Add
  - `VUID-VkAttachmentDescription-format-06242`
  - `VUID-VkAttachmentDescription-format-06243`
  - `VUID-VkAttachmentDescription2-format-06249`
  - `VUID-VkAttachmentDescription2-format-06250`
  - `VUID-VkAttachmentDescription-synchronization2-06908`
  - `VUID-VkAttachmentDescription-synchronization2-06909`
  - `VUID-VkAttachmentDescription2-synchronization2-06908`
  - `VUID-VkAttachmentDescription2-synchronization2-06909`

### CODE REVIEW

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp [hunk 1]
```diff
@@ -1191,18 +1234,17 @@ TEST_F(VkLayerTest, RenderPassCreateAttachmentReferenceInvalidSync2Layout) {
     TEST_DESCRIPTION("Attachment reference uses sync2 and ATTACHMENT_OPTIMAL_KHR or READ_ONLY_OPTIMAL_KHR layouts");
 
     SetTargetApiVersion(VK_API_VERSION_1_3);
-
+    AddRequiredExtensions(VK_KHR_CREATE_RENDERPASS_2_EXTENSION_NAME);
```
Comment (jeremyg-lunarg) [inline tests/vklayertests_descriptor_renderpass_framebuffer.cpp:1237]: this tests requires renderpass2 and vulkan 1.3. Couldn't it just require 1.3?

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM / +2

File: layers/parameter_validation_utils.cpp [hunk 2]
```diff
@@ -4504,6 +4504,447 @@ bool StatelessValidation::manual_PreCallValidateUpdateDescriptorSets(VkDevice de
...
+                skip |= LogError(device, vuid,
+                                 "%s: pCreateInfo->pAttachments[%" PRIu32
+                                 "].finalLayout must not be "
+                                 "VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL or "
+                                 "VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_OPTIMAL",
+                                 func_name, i);
+            }
+        }
+        if (attachment_description_stencil_layout) {
+            if (attachment_description_stencil_layout->stencilInitialLayout == VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL ||
```
Comment (juan-lunarg) [inline layers/parameter_validation_utils.cpp:4754]: Create helper variables. Consistent with above code and helps readability.

```suggestion
            const VkImageLayout stencilInitialLayout = attachment_description_stencil_layout->stencilInitialLayout;
            const VkImageLayout stencilFinalLayout = attachment_description_stencil_layout->stencilFinalLayout;
            if (stencilInitialLayout == VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL ||
```

File: layers/parameter_validation_utils.cpp [hunk 3]
```diff
@@ -4504,6 +4504,447 @@ bool StatelessValidation::manual_PreCallValidateUpdateDescriptorSets(VkDevice de
     return validate_WriteDescriptorSet("vkUpdateDescriptorSets", descriptorWriteCount, pDescriptorWrites, false);
 }
 
+template <typename RenderPassCreateInfoGeneric>
+bool StatelessValidation::CreateRenderPassGeneric(VkDevice device, const RenderPassCreateInfoGeneric *pCreateInfo,
+                                                  const VkAllocationCallbacks *pAllocator, VkRenderPass *pRenderPass,
+                                                  RenderPassCreateVersion rp_version) const {
+    bool skip = false;
+    uint32_t max_color_attachments = device_limits.maxColorAttachments;
+    bool use_rp2 = (rp_version == RENDER_PASS_VERSION_2);
```
Comment (juan-lunarg) [inline layers/parameter_validation_utils.cpp:4513]:
```suggestion
    const bool use_rp2 = (rp_version == RENDER_PASS_VERSION_2);
```

File: layers/parameter_validation_utils.cpp [hunk 4]
```diff
@@ -4504,6 +4504,447 @@ bool StatelessValidation::manual_PreCallValidateUpdateDescriptorSets(VkDevice de
...
+    const char *func_name = (use_rp2) ? "vkCreateRenderPass2" : "vkCreateRenderPass";
+    const char *vuid;
```
Comment (juan-lunarg) [inline layers/parameter_validation_utils.cpp:4515]:
```suggestion
    const char *vuid = nullptr;
```

**Review by juan-lunarg — CHANGES_REQUESTED** [review]
minor nitpicks

File: layers/parameter_validation_utils.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply layers/parameter_validation_utils.cpp:4754]: This part was not touched in the copy over, but agree should be approved while here!

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply tests/vklayertests_descriptor_renderpass_framebuffer.cpp:1237]: yes, but `TestRenderPassCreate` had a bug, fixed and removed this `AddRequiredExtensions`

**Review by juan-lunarg — APPROVED** [review]

**Pull request merged**